x-pack/legacy/plugins/security/server/routes/api/v1/authenticate.js

```diff
@@ -82,6 +83,29 @@ export function initAuthenticateApi({ authc: { login, logout }, config }, server
     }
   });
 
+  server.route({
+    method: 'GET',
+    path: '/api/security/v1/oidc/implicit',
+    config: { auth: false },
+    async handler(request, h) {
+      const legacyConfig = server.config();
+      const basePath = legacyConfig.get('server.basePath');
+
+      const nonce = await generateCSPNonce();
+      const cspRulesHeader = createCSPRuleString(legacyConfig.get('csp.rules'), nonce);
```

Comment:
__note:__ using CSP here feels like an overkill here, but it doesn't hurt to have it I think.

Reply:
Do we need to do anything about `server.customResponseHeaders` explicitly, or does the global hapi configuration apply automatically? 

I've been trying to think of potential issues that we get from rendering all of the HTML ourselves, but I haven't been able to think of any.

Reply:
> Do we need to do anything about server.customResponseHeaders explicitly, or does the global hapi configuration apply automatically?

Hapi's `onPreResponse` hooks do everything for us "automagically". I actually was surprised that CSP isn't done via `onPreResponse` too... But I admit I don't know much about all the variables we had to account for when we were implementing it.